EasyReflectometry/sample/assemblies/surfactant_layer.py

```diff
@@ -226,8 +237,8 @@ def _dict_repr(self) -> dict:
         :return: Simple dictionary
         """
         return {
-            'layer1': self.layers[0]._dict_repr,
-            'layer2': self.layers[1]._dict_repr,
+            'top_layer': self.top_layer._dict_repr,
```

Comment:
`gradient_layer` also returns `type: self._type` in this serializer. Any reason to not include it here?

Reply:
Alternatively, I can see `type` is not really used for anything in any of the derived classes (or in the base class).
Maybe it's ok to just get rid of it?

Reply:
Tried to remove it but got an error in the fitting test.  Seems like EasyCore relies on this property.  
- Added code comment
- Removed it from the `_dict_repr` in `gradient_layer`
- Added when calling the super init in repeating multi layer